Turn user task `candidateUsers` and `candidateGroups` into lists

tmetzke opened the pull request:
## Description

* Removes the attributes `candidateUsers` and `candidateGroups` from the records.
* Introduces the attributes `candidateUsersList` and `candidateGroupsList` of type `ArrayProperty<String>`. 
* Parses the expressions of the candidate attributes as before but stores them as lists directly instead of converting them to list literals.
* Reflects the changes in related test and serialization code.

## ⚠️ BREAKING CHANGES

* The values of the `candidateUsers` and `candidateGroups` attributes in `UserTaskRecord`s are lost during an update from `8.4` to `8.5`.

## Related issues

<!-- Which issues are closed by this PR or are related -->

closes #16007 

## Definition of Done

<!-- Please check the items that apply, before merging or (if possible) before requesting a review. -->

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

korthout commented on protocol/revapi.json:
```diff
@@ -336,6 +336,54 @@
           "code": "java.method.removed",
           "old":  "method io.camunda.zeebe.protocol.record.value.ImmutableCompensationSubscriptionRecordValue io.camunda.zeebe.protocol.record.value.ImmutableCompensationSubscriptionRecordValue::withCompensableActivityScopeId(long)",
           "justification": "This functionality is not yet released"
+        },
+        {
+          "ignore": true,
+          "code": "java.method.removed",
+          "old":  "method io.camunda.zeebe.protocol.record.value.ImmutableUserTaskRecordValue.Builder io.camunda.zeebe.protocol.record.value.ImmutableUserTaskRecordValue.Builder::withCandidateGroups(java.lang.String)",
+          "justification": "This functionality is not yet released"
+        },
```
❌ Please add such ignores to `ignored-changes.json` so they can be cleared on release: https://github.com/camunda/zeebe/wiki/Breaking-Changes#java

It could also benefit from a more thorough justification, perhaps linking your analysis of the breaking change

tmetzke replied:
Thanks, @korthout, I wasn't aware of those docs, glad we have them 👍 
I'll add this 🙂

tmetzke replied:
Just be certain: we add such changes twice now, in the `revapi.json` and also in an `ignored-changes.json` file?

tmetzke replied:
Addressed with https://github.com/camunda/zeebe/pull/16141/commits/cc195fff780ceb3500cc637993548ec5e2c8c4fa

korthout replied:
>we add such changes twice now, in the revapi.json and also in an ignored-changes.json file?

@tmetzke No, only in the `ignored-changes.json` file. The `revapi.json` file is for long term configurations, while the ignored changes are cleared with every release.

tmetzke replied:
Thanks, @korthout, that makes a lot of sense. 👍 
Looking at [this PR](https://github.com/camunda/zeebe/pull/13068) as an example (there are more like it), we added similar entries only to the `revapi,json`. We did not create the `ignored-changes.json` or add entries there. Thus I am wondering if we either
1. Need to make the team aware again of how this is done.
2. Need to adjust the documentation and/or processes to reflect how we do this now.

korthout replied:
That's a good idea @tmetzke. Let's check first if there is some reason I'm missing.

@Zelldon do you remember why [these revapi ignores](https://github.com/camunda/zeebe/pull/13068/files#diff-92a55e36a54aca257697d944be71f135bc5e2644a351e822545283a1df1652deR186) were added to `revapi.json` rather than `ignored-changes.json`?

korthout reviewed (approved):
Nice work @tmetzke 👍 These are the changes I expected 🙇 

❌ I have just one change request regarding the revapi config. However, I won't require an additional review.

So, after that change, this LGTM 👍 

EDIT: Please also ensure to document the breaking change in the update guide

tmetzke commented:
New changes:
* Rebased onto the main branch.
* Adjusted the justifications in the revapi configuration.
* Moved the changes from `revapi.json` to `ignored-changes.json`.

tmetzke commented:
@korthout, if we're good with the PR here as it is, I'll go ahead and merge it. I've adjusted the Revapi configuration.
Regarding the follow-ups discussed in the open conversation, I propose we take them to a separate issue to make progress here. What do you think?
